distfunc.c

```diff
@@ -0,0 +1,153 @@
+#include "postgres.h"
+#include "embedding.h"
+#include "math.h"
+
+#ifdef __x86_64__
+#include <immintrin.h>
+
+__attribute__((target("avx2")))
+static dist_t l2_dist_impl_avx2(const coord_t *x, const coord_t *y, size_t n)
```

Comment:
This works. Alternatively you can just copy-paste the scalar version and mark it with `__attribute__((target("avx2")))` and it will autovectorize it.